Lacrit/assignment1/q2.cpp

```diff
@@ -4,44 +4,76 @@
 template<class T> 
 struct Node
 {
-    T i;
-    struct Node *next;
+	private:
+    	T i;
+		Node *next;
+	public: 
+		Node* get_next()
+		{
+			return next;
+		}  
+		T get_i()
+		{
+			return i;
+		}
+		void set_next(Node *new_next)
+		{
+			next = new_next;
+		}
+		void set_i(T j)
+		{
+			i = j;
+		}
+			
 };
 
 
 template<class T> 
 T findNode (Node<T> *listHead, int n)
 {
+	 if (listHead == NULL) return 0; 
```

Comment:
nullptr is usually preferred to NULL, since nullptr is typed, while NULL is a int with value 0
Also, returning 0 severely limits the types T can be :)